File: study/ts/script/scene/view/cakeActivity/CakeRankPlayerCell.ts
import ListViewCellBase from "../../../ui/ListViewCellBase";
import { CakeActivityDataHelper } from "../../../utils/data/CakeActivityDataHelper";
import { Path } from "../../../utils/Path";
import UIHelper from "../../../utils/UIHelper";
import { TextHelper } from "../../../utils/TextHelper";

const { ccclass, property } = cc._decorator;

@ccclass
export default class CakeRankPlayerCell extends ListViewCellBase {

    @property({
        type: cc.Node,
        visible: true
    })
    _panelBg: cc.Node = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textRank: cc.Label = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textServerName: cc.Label = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textName: cc.Label = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textScore: cc.Label = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageArrow: cc.Sprite = null;

    onCreate() {
        var size = this._panelBg.getContentSize();
        this.node.setContentSize(size.width, size.height);
        this._imageArrow.sizeMode = cc.Sprite.SizeMode.RAW;
    }
    updateUI(data) {
        this._textRank.string = (data.getRank());
        var serverName =  TextHelper.cutText(data.getServer_name(), 5);
        this._textServerName.string = (serverName);
        this._textName.string = (data.getName());
        this._textScore.string = (data.getPoint());
        UIHelper.loadTexture(this._imageArrow, Path.getUICommon(data.getChangeResName()))
    }

}

File: study/lua/src/app/scene/view/cakeActivity/CakeRankPlayerCell.lua
--
-- Author: Liangxu
-- Date: 2019-4-30
-- 蛋糕活动玩家排行榜Cell

local ListViewCellBase = require("app.ui.ListViewCellBase")
local CakeRankPlayerCell = class("CakeRankPlayerCell", ListViewCellBase)
local CakeActivityDataHelper = require("app.utils.data.CakeActivityDataHelper")

function CakeRankPlayerCell:ctor()
	local resource = {
		file = Path.getCSB("CakeRankPlayerCell", "cakeActivity"),
		binding = {
			
		}
	}
	CakeRankPlayerCell.super.ctor(self, resource)
end

function CakeRankPlayerCell:onCreate()
	local size = self._panelBg:getContentSize()
	self:setContentSize(size.width, size.height)
	self._imageArrow:ignoreContentAdaptWithSize(true)
end

function CakeRankPlayerCell:update(data)
	self._textRank:setString(data:getRank())
	local serverName = require("app.utils.TextHelper").cutText(data:getServer_name(), 5)
	self._textServerName:setString(serverName)
	self._textName:setString(data:getName())
	self._textScore:setString(data:getPoint())
	self._imageArrow:loadTexture(Path.getUICommon(data:getChangeResName()))
end

return CakeRankPlayerCell
